private/protocol/query/unmarshal_error.go

```diff
@@ -36,5 +45,22 @@ func UnmarshalError(r *request.Request) {
 			r.HTTPResponse.StatusCode,
 			reqID,
 		)
+		return
+	}
+
+	// Check for unhandled error
+	servUnavailResp := xmlServiceUnavailableResponse{}
+	unavailErr := xml.Unmarshal(bodyBytes, &servUnavailResp)
```

Comment:
Removing the name tag all together might work if the only tags were `ErrorResponse` or `ServiceUnavailableException` i think. If there were any other tag before the two error tags that we are expecting I think the xml will not unmarshal the error we're actually looking for.  The additional unmarshal should be an extremely rare event and only occurs if ErrorResponse failed to parse the error body.